core/dragged_connection_manager.js

```diff
@@ -0,0 +1,208 @@
+ */
+Blockly.DraggedConnectionManager.prototype.update = function(e, dxy) {
+  var oldClosestConnection = this.closestConnection_;
+  var closestConnectionChanged = this.updateClosest_(dxy);
+
+  if (closestConnectionChanged) {
+    if (oldClosestConnection) {
+      oldClosestConnection.unhighlight();
+    }
+  }
+
+  var wouldDeleteBlock = this.updateCursor_(e);
+
+  if (!wouldDeleteBlock && closestConnectionChanged &&
+      this.closestConnection_) {
+    this.addConnectionHighlighting();
+  }
+};
+
+/**
+ * Remove highlighting from the currently highlighted connection, if it exists.
+ */
+Blockly.DraggedConnectionManager.prototype.removeConnectionHighlighting = function() {
+  if (this.closestConnection_) {
+    this.closestConnection_.unhighlight();
+  }
+};
+
+/**
+ * Add highlighting to the closest connection, if it exists.
+ */
+Blockly.DraggedConnectionManager.prototype.addConnectionHighlighting = function() {
+  if (this.closestConnection_) {
+    this.closestConnection_.highlight();
+  }
+};
+
+/**
+ * Populate the list of available connections on this block stack.  This should
+ * only be called once, at the beginning of a drag.
+ * @private
+ */
+Blockly.DraggedConnectionManager.prototype.setAvailableConnections_ = function() {
+  this.availableConnections_ = this.topBlock_.getConnections_(false);
+  // Also check the last connection on this stack
+  var lastOnStack = this.topBlock_.lastConnectionInStack_();
+  if (lastOnStack && lastOnStack != this.topBlock_.nextConnection) {
+    this.availableConnections_.push(lastOnStack);
+  }
+};
+
+/**
+ * Find the new closest connection, and update internal state in response.
+ * @param {!goog.math.Coordinate} dxy Position relative to the drag start.
+ * @return {boolean} Whether the closest connection has changed.
+ * @private
+ */
+Blockly.DraggedConnectionManager.prototype.updateClosest_ = function(dxy) {
+  var oldClosestConnection = this.closestConnection_;
+
+  this.closestConnection_ = null;
+  this.localConnection_ = null;
+  this.radiusConnection_ = Blockly.SNAP_RADIUS;
+  for (var i = 0; i < this.availableConnections_.length; i++) {
+    var myConnection = this.availableConnections_[i];
+    var neighbour = myConnection.closest(this.radiusConnection_, dxy);
+    if (neighbour.connection) {
+      this.closestConnection_ = neighbour.connection;
+      this.localConnection_ = myConnection;
+      this.radiusConnection_ = neighbour.radius;
+    }
+  }
+  return oldClosestConnection != this.closestConnection_;
+};
+
+/**
+ * Provide visual indication of whether the block will be deleted if
+ * dropped here.
+ * Prefer connecting over dropping into the trash can, but prefer dragging to
+ * the toolbox over connecting to other blocks.
+ * @param {!Event} e Mouse move event.
+ * @return {boolean} True if the block would be deleted if dropped here,
+ *     otherwise false.
+ * @private
+ */
+Blockly.DraggedConnectionManager.prototype.updateCursor_ = function(e) {
```

Comment:
It doesn't fit anywhere.  ¯\_(°_°)_/¯

And it does more than updating the cursor.  (╯°□°）╯︵ ┻━┻

We had the same discussion when I originally wrote the function--it's just doing too many tightly connected things.  This is another one that I'm hoping I can untangle in my dragging work.